test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.server.cs

```diff
@@ -62,11 +78,30 @@ public void GrpcAspNetCoreInstrumentationAddsCorrectAttributes()
                 invo.Method.Name == "OnEnd" && (invo.Arguments[0] as Activity).OperationName == "Microsoft.AspNetCore.Hosting.HttpRequestIn").Arguments[0];
 
             Assert.Equal(ActivityKind.Server, activity.Kind);
-            Assert.Equal("grpc", activity.GetTagValue(SemanticConventions.AttributeRpcSystem));
-            Assert.Equal("greet.Greeter", activity.GetTagValue(SemanticConventions.AttributeRpcService));
-            Assert.Equal("SayHello", activity.GetTagValue(SemanticConventions.AttributeRpcMethod));
-            Assert.Contains(activity.GetTagValue(SemanticConventions.AttributeNetPeerIp), clientLoopbackAddresses);
-            Assert.NotEqual(0, activity.GetTagValue(SemanticConventions.AttributeNetPeerPort));
+
+            if (!enableGrpcAspNetCoreSupport.HasValue || enableGrpcAspNetCoreSupport.Value)
+            {
+                Assert.Equal("grpc", activity.GetTagValue(SemanticConventions.AttributeRpcSystem));
+                Assert.Equal("greet.Greeter", activity.GetTagValue(SemanticConventions.AttributeRpcService));
+                Assert.Equal("SayHello", activity.GetTagValue(SemanticConventions.AttributeRpcMethod));
+                Assert.Contains(activity.GetTagValue(SemanticConventions.AttributeNetPeerIp), clientLoopbackAddresses);
+                Assert.NotEqual(0, activity.GetTagValue(SemanticConventions.AttributeNetPeerPort));
+
+                // Tags added by the library then removed from the instrumentation
+                // TODO: The grpc.status_code attribute added by the library is not currently
+                // removed because the tracing spec for span status is no longer based on
+                // gRPC status codes. Ultimately, the grpc.status_code tag should be replaced
+                // by whatever semantic convention is settled on in the RPC spec.
+                // See: https://github.com/open-telemetry/opentelemetry-dotnet/issues/1345
+                Assert.Null(activity.GetTagValue(GrpcTagHelper.GrpcMethodTagName));
+                Assert.NotNull(activity.GetTagValue(GrpcTagHelper.GrpcStatusCodeTagName));
+            }
+            else
+            {
+                Assert.NotNull(activity.GetTagValue(GrpcTagHelper.GrpcMethodTagName));
```

Comment:
as these are added *not* by Otel Instrumentation, do we need to validate this?

Reply:
My original thought was that since the instrumentation removes these tags when this option is enabled it made sense to me to validate that the instrumentation left them untouched when when the option is disabled.